lib/network/modules/components/shared/ComponentUtil.js

```diff
@@ -49,6 +65,56 @@ class ComponentUtil {
 
     return value;
   }
+
+
+  /**
+   * Check if the point falls within the given rectangle.
+   *
+   * @param {rect} rect
+   * @param {point} point
+   * @param {rotationPoint} [rotationPoint] if specified, the rotation that applies to the rectangle.
+   * @returns {boolean}  true if point within rectangle, false otherwise
+   * @static
+   */
+  static pointInRect(rect, point, rotationPoint) {
+    if (rect.right <= rect.left || rect.bottom <= rect.top) {
+      return false;  // early out
+    }
+
+    if (rotationPoint !== undefined) {
+      // Rotate the point the same amount as the rectangle
+      var tmp = {
```

Comment:
Can we rename all tmp variables to more descriptive names, like `candidatePoint` or `rotatedPoint`?

Reply:
For small scopes I prefer short names, I believe this enhances readability. Also, `tmp` has meaning here; it's a clear indication that this is meant as a temporary variable.

Reply:
I won't fight you on this, but rather agree to disagree.

Reply:
ok